server/app/routes.py

```diff
@@ -431,3 +431,133 @@ def delete_invitation(invitation_id):
+    )
+    db.session.add(new_inventory)
+    db.session.commit()
+    return jsonify({'message': 'New inventory item created!'}), 201
+
+@app.route('/inventory/<int:inventory_id>', methods=['PUT'])
+def update_inventory(inventory_id):
+    data = request.get_json()
+    item = Inventory.query.get_or_404(inventory_id)
+    item.product_id = data['product_id']
+    item.store_id = data['store_id']
+    item.quantity_received = data['quantity_received']
+    item.quantity_in_stock = data['quantity_in_stock']
+    item.quantity_spoilt = data['quantity_spoilt']
+    item.payment_status = data['payment_status']
+    db.session.commit()
+    return jsonify({'message': 'Inventory item updated!'})
+
+@app.route('/inventory/<int:inventory_id>', methods=['DELETE'])
+def delete_inventory(inventory_id):
+    item = Inventory.query.get_or_404(inventory_id)
+    db.session.delete(item)
+    db.session.commit()
+    return jsonify({'message': 'Inventory item deleted!'})
+
+# Supply Request Routes
+
+@app.route('/supply_requests', methods=['GET'])
+def get_all_supply_requests():
+    requests = SupplyRequest.query.all()
+    result = []
+    for req in requests:
+        result.append({
+            'request_id': req.request_id,
+            'inventory_id': req.inventory_id,
+            'user_id': req.user_id,
+            'request_date': req.request_date,
+            'status': req.status
+        })
+    return jsonify(result)
+
+@app.route('/supply_requests/<int:request_id>', methods=['GET'])
+def get_supply_request(request_id):
+    req = SupplyRequest.query.get_or_404(request_id)
+    result = {
+        'request_id': req.request_id,
+        'inventory_id': req.inventory_id,
+        'user_id': req.user_id,
+        'request_date': req.request_date,
+        'status': req.status
+    }
+    return jsonify(result)
+
+@app.route('/supply_requests', methods=['POST'])
+def create_supply_request():
+    data = request.get_json()
+    new_request = SupplyRequest(
+        inventory_id=data['inventory_id'],
+        user_id=data['user_id'],
+        request_date=datetime.utcnow(),
+        status=data['status']
+    )
+    db.session.add(new_request)
+    db.session.commit()
+    return jsonify({'message': 'New supply request created!'}), 201
+
+@app.route('/supply_requests/<int:request_id>', methods=['PUT'])
+def update_supply_request(request_id):
+    data = request.get_json()
+    req = SupplyRequest.query.get_or_404(request_id)
+    req.inventory_id = data['inventory_id']
+    req.user_id = data['user_id']
+    req.request_date = data['request_date']
+    req.status = data['status']
+    db.session.commit()
+    return jsonify({'message': 'Supply request updated!'})
+
+@app.route('/supply_requests/<int:request_id>', methods=['DELETE'])
+def delete_supply_request(request_id):
+    req = SupplyRequest.query.get_or_404(request_id)
+    db.session.delete(req)
+    db.session.commit()
+    return jsonify({'message': 'Supply request deleted!'})
```

Comment:
Keep the response format consistent so that it maintains a consistent structure.
Use this as the reference:
```python
return jsonify({
            "status": "Failed",
            "message": "Email and user_id are required.",
            "data": None
        }), 400
```